### PR TITLE
[API Compatibility] add paddle.compat.nn.functional.log_softmax -part

### PR DESCRIPTION
**Pull request by Manfredss**
### PR Category
User Experience

### PR Types
New features

### Description
align paddle.nn.functional.log_softmax
add paddle.compact.nn.functional.log_softmax

### 是否引起精度变化
否

### CODE REVIEW

File: python/paddle/tensor/__init__.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
     real,
     shape,
 )
+from .compat_log_softmax import log_softmax as log_softmax
```
Comment (zhwesky2010) [inline python/paddle/tensor/__init__.py]: 直接放到compat_softmax文件里吧，减少下文件。

这三个一起对齐：
```
torch.nn.functional.log_softmax -> paddle.compat.nn.functional.log_softmax
torch.log_softmax -> paddle.compat.nn.functional.log_softmax的别名
torch.special.log_softmax -> paddle.compat.nn.functional.log_softmax的别名
torch.Tensor.log_softmax 
```

File: python/paddle/tensor/compat_log_softmax.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    illegal_keys={"x", "axis", "name"},
+    ignore_param=('_stacklevel', 2, int),
+    func_name="paddle.compat.nn.functional.log_softmax",
+    correct_name="paddle.nn.functional.log_softmax",
+    url_suffix="torch.nn.functional.log_softmax",
+)
+def log_softmax(
+    input: Tensor,
+    dim: int | None = None,
+    dtype: DTypeLike | None = None,
```
Comment (zhwesky2010) [inline python/paddle/tensor/compat_log_softmax.py]: 补一个out参数：

`torch.log_softmax(Tensor input, int dim, torch.dtype dtype = None, *, Tensor out = None)`

File: python/paddle/nn/functional/activation.py [hunk 3]
```diff
@@ -1696,6 +1696,7 @@ def thresholded_relu_(
         return _C_ops.thresholded_relu_(x, threshold, value)
 
 
+@param_two_alias(["x", "input"], ["axis", "dim"])
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/activation.py:1699]: 这个也补个out参数，格式更一致

File: python/paddle/tensor/__init__.py [hunk 4]
```diff
@@ -31,7 +31,7 @@
     real,
     shape,
 )
-from .compat_softmax import softmax as softmax
+from .compat_softmax import log_softmax as log_softmax, softmax as softmax
```
Comment (zhwesky2010) [inline python/paddle/tensor/__init__.py]: 这里是不是不用as，其他都是直接import进来

File: test/legacy_test/test_compat_log_softmax.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+# limitations under the License.
+import unittest
+
+import numpy as np
+
+import paddle
+import paddle.nn.functional as F
+
+
+class TestCompatLogSoftmaxCorrectness(unittest.TestCase):
```
Comment (zhwesky2010) [inline test/legacy_test/test_compat_log_softmax.py]: `paddle.log_softmax、paddle.Tensor.log_softmax、paddle.special.log_softmax、paddle.compat.nn.functional.log_softmax`都测一下pytorch的用法，可以同步完善PaConvert的测试，并将这些兼容性测试搬到paddle repo里来。

兼容性测试规范：

Image

File: python/paddle/nn/functional/activation.py [hunk 6]
```diff
@@ -1696,11 +1696,14 @@ def thresholded_relu_(
...
 def log_softmax(
     x: Tensor,
     axis: int = -1,
     dtype: DTypeLike | None = None,
     name: str | None = None,
+    *,
+    out: Tensor | None = None,
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/activation.py:1706]: out没有实际处理，`paddle.nn.functional.softmax`也测下，一共测5个API

File: python/paddle/tensor/compat_softmax.py [hunk 7]
```diff
@@ -178,3 +178,84 @@ def softmax(
...
+    if (
+        (dtype is not None)
+        and (not isinstance(dtype, core.VarDesc.VarType))
+        and (not isinstance(dtype, core.DataType))
+    ):
+        dtype = convert_np_dtype_to_dtype_(dtype)
+
+    if in_dynamic_or_pir_mode():
+        outs_cast = input if dtype is None else _C_ops.cast(input, dtype)
+        return _C_ops.log_softmax(outs_cast, dim)
```
Comment (zhwesky2010) [inline python/paddle/tensor/compat_softmax.py]: out测试了没有

File: python/paddle/__init__.py [hunk 8]
```diff
@@ -319,7 +319,9 @@ def new_init(self, *args, **kwargs):
     real,
     shape,
 )
-from .tensor.compat_softmax import softmax
+from .tensor.compat_softmax import log_softmax, softmax
+
+framework.core.eager.Tensor.log_softmax = log_softmax
```
Comment (zhwesky2010) [inline python/paddle/__init__.py]: 放到tensor_method_func这个list里，会自动patch，无需手动patch

File: test/legacy_test/test_compat_log_softmax.py [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+
+        # 4. PyTorch keyword arguments (alias)
+        out4 = F.log_softmax(input=x, dim=-1)
+
+        # 5. Mixed arguments (positional + keyword)
+        out5 = F.log_softmax(x, dim=-1)
+        out5b = F.log_softmax(x, -1, dtype=None)
+
+        # 6. out parameter (accepted for API compat)
+        out6 = F.log_softmax(x, -1, out=paddle.empty_like(x))
```
Comment (zhwesky2010) [inline test/legacy_test/test_compat_log_softmax.py]: 这里不能这么测，要这样测：
```
out6 = paddle.empty_like(x)
F.log_softmax(x, -1, out=out6)
```

File: test/legacy_test/test_log_softmax.py [hunk 10]
```diff
@@ -291,6 +291,105 @@ def test_check_grad(self):
         self.check_grad(['X'], ['Out'], check_pir=True)
 
 
+class TestLogSoftmaxParamAlias(unittest.TestCase):
```
Comment (zhwesky2010) [inline test/legacy_test/test_log_softmax.py:294]: 测下out

**Comment by zhwesky2010** [comment]
@Manfredss 测试完全无法通过，查看CI日志

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by SigureMo — APPROVED** [review]
